modules/contentpool_channel_remote/contentpool_channel_remote.module.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/**
+ * @file
+ * Custom elements hooks.
+ */
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function contentpool_channel_remote_form_taxonomy_term_channel_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $term = $form_state->getFormObject()->getEntity();
+
+  if (!empty($term) && isset($form['field_remote_site'])) {
+    // Check if term is top level; if not hide field_remote_site.
+    if ($term->parent->target_id) {
+      $form['field_remote_site']['#access'] = FALSE;
+    }
+  }
+}
+
+/**
+ * Implements hook_entity_presave().
+ *
+ * Remove remote_site reference if term is no longer top level.
+ */
+function contentpool_channel_remote_entity_presave(EntityInterface $entity) {
```

Comment:
minor: it's generally nicer (and faster) to implement hook_ENTITY_TYPE_ID_presave() instead of this + the if.

Reply:
Don't think those hooks are triggered if term is changed from term overview page (eg. moved under a parent). Tried both hook_ENTITY_TYPE_ID_presave()  and hook_ENTITY_TYPE_ID_update()

Reply:
I see, that sucks. This means the value will stay in this case. Well we'll just have to accept that and keep in mind to check the term depth in addition to the field value when we use it.